core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/FixedRateSamplingTelemetryProcessor.java

```diff
@@ -41,14 +47,17 @@
 @BuiltInProcessor("FixedRateSamplingTelemetryProcessor")
 public final class FixedRateSamplingTelemetryProcessor implements TelemetryProcessor {
 
-    private final String dependencyTelemetryName = "Dependency";
-    private static final String eventTelemetryName = "Event";
-    private static final String exceptionTelemetryName = "Exception";
-    private static final String pageViewTelemetryName = "PageView";
-    private static final String requestTelemetryName = "Request";
-    private static final String traceTelemetryName = "Trace";
+    public static final double DEFAULT_SAMPLING_PERCENTAGE = 100.;
+    private static Map<TelemetryType, Class> allowedTypes = new HashMap<>();
 
-    private static Map<String, Class> allowedTypes;
+    static {
```

Comment:
I'm fine with moving this to the static constructor, but changing the map type is unnecessary. We don't need to introduce parsing into this. String.hashCode does a fine job here.

Reply:
By doing this IDE can perform auto-complete when users are typing this properties. That was the reason of this change.

Reply:
@dhaval24, let's discuss offline

Reply:
@littleaj I have given some deep thoughts on this, and I think I would agree with you on this. Here are the reasons: This enum is only used for the sampling, not required elsewhere. It adds an overhead of maintaining it and ensuring that all new types are also added to it. A person can easily miss this unless there is an automated way to do it. Further more this enum is public which then adds the overhead of not deprecating it before a major version. The degree of advantage it provides is not absolute necessity and overhead is pretty high.

I will make this changes :) Thanks for drawing the attention.

Reply:
#resolved